gapis/gfxapi/gles/gles.api

```diff
@@ -515,34 +515,47 @@ sub void ApplyStaticContextState(ref!Context ctx, ref!StaticContextState staticS
 
 sub void ApplyDynamicContextState(ref!Context ctx, ref!DynamicContextState dynamicState) {
   if (dynamicState != null) {
-    backbuffer := ctx.Objects.Framebuffers[0]
-    color_id := backbuffer.ColorAttachments[0].Renderbuffer.ID
-    depth_id := backbuffer.DepthAttachment.Renderbuffer.ID
-    stencil_id := backbuffer.StencilAttachment.Renderbuffer.ID
-
-    color_buffer := ctx.SharedObjects.Renderbuffers[color_id]
-    color_buffer.Width = dynamicState.BackbufferWidth
-    color_buffer.Height = dynamicState.BackbufferHeight
-    color_buffer.InternalFormat = dynamicState.BackbufferColorFmt
-    ctx.SharedObjects.Renderbuffers[color_id] = color_buffer
-
-    depth_buffer := ctx.SharedObjects.Renderbuffers[depth_id]
-    depth_buffer.Width = dynamicState.BackbufferWidth
-    depth_buffer.Height = dynamicState.BackbufferHeight
-    depth_buffer.InternalFormat = dynamicState.BackbufferDepthFmt
-    ctx.SharedObjects.Renderbuffers[depth_id] = depth_buffer
-
-    stencil_buffer := ctx.SharedObjects.Renderbuffers[stencil_id]
-    stencil_buffer.Width = dynamicState.BackbufferWidth
-    stencil_buffer.Height = dynamicState.BackbufferHeight
-    stencil_buffer.InternalFormat = dynamicState.BackbufferStencilFmt
-    ctx.SharedObjects.Renderbuffers[stencil_id] = stencil_buffer
-
-    if dynamicState.ResetViewportScissor {
-      ctx.FragmentOperations.Scissor.Box.Width = dynamicState.BackbufferWidth
-      ctx.FragmentOperations.Scissor.Box.Height = dynamicState.BackbufferHeight
-      ctx.Rasterization.Viewport.Width = dynamicState.BackbufferWidth
-      ctx.Rasterization.Viewport.Height = dynamicState.BackbufferHeight
+    if 0 in ctx.Objects.Framebuffers {
```

Comment:
glGenFramebuffers returning 0's is a bit less scary since we want to handle invalid traces anway, and it means I do not have check the whole code-base on where else I have (correctly) assumed the default framebuffer has attachments (and all other similar assumptions).

Nonetheless, I would still argue that hardening of the API against invalid input should be done by tests and fuzzers, not by race condition on user machines with valid traces.

> now running all state mutations on a single executor
Well, I phrased it in single-threaded, but the logic applies to multi-threaded.  I assume we have a semaphore somewhere to limit the number of concurrent tasks to the number of CPUs, right?